File: BackEnd/src/main/java/com/Checkerly/BackEnd/dto/EventDTO.java
package com.Checkerly.BackEnd.dto;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalTime;
import java.util.Date;

import org.springframework.data.annotation.Id;

import com.Checkerly.BackEnd.domain.Event;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor

public class EventDTO implements Serializable{
   	@Serial
	private static final long serialVersionUID = 1L;
    @Id
	private String id;
	private String nomeEvento;
	private String assuntoEvento;
	private Double latitude;
	private Double longitude;
	private Date dataInicio;
	private Date dataFim;
    private LocalTime horaEvento;

    public EventDTO(Event obj) {
		id = obj.getId();
		nomeEvento = obj.getNomeEvento();
		assuntoEvento = obj.getAssuntoEvento();
		latitude = obj.getLatitude();
		longitude = obj.getLongitude();
		dataInicio = obj.getDataInicio();
		dataFim = obj.getDataFim();
        horaEvento = obj.getHoraEvento();
	}

}

File: BackEnd/src/main/kotlin/com/Checkerly/BackEnd/dto/EventDTO.kt
package com.Checkerly.BackEnd.dto

import com.Checkerly.BackEnd.domain.Event
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter
import org.springframework.data.annotation.Id
import java.io.Serial
import java.io.Serializable
import java.time.LocalTime
import java.util.Date

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
class EventDTO(obj: Event) : Serializable {
    @Id
    private val id: String?
    private val nomeEvento: String?
    private val assuntoEvento: String?
    private val latitude: String?
    private val longitude: String?
    private val dataInicio: Date?
    private val dataFim: Date?
    private val horaEvento: LocalTime?

    init {
        id = obj.getId()
        nomeEvento = obj.getNomeEvento()
        assuntoEvento = obj.getAssuntoEvento()
        latitude = obj.getLatitude()
        longitude = obj.getLongitude()
        dataInicio = obj.getDataInicio()
        dataFim = obj.getDataFim()
        horaEvento = obj.getHoraEvento()
    }

    companion object {
        @Serial
        private const val serialVersionUID: Long = 1L
    }
}
